google/resource_google_service_account_key.go

```diff
@@ -0,0 +1,143 @@
+package google
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/encryption"
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/iam/v1"
+)
+
+func resourceGoogleServiceAccountKey() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGoogleServiceAccountKeyCreate,
+		Read:   resourceGoogleServiceAccountKeyRead,
+		Delete: resourceGoogleServiceAccountKeyDelete,
+		Schema: map[string]*schema.Schema{
+			"service_account_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"key_algorithm": &schema.Schema{
+				Type:     schema.TypeString,
+				Default:  "KEY_ALG_RSA_2048",
+				Optional: true,
+				ForceNew: true,
+			},
+			"private_key": &schema.Schema{
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"private_key_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Default:  "TYPE_GOOGLE_CREDENTIALS_FILE",
+				Optional: true,
+				ForceNew: true,
+			},
+			"valid_after": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"valid_before": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"pgp_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"private_key_encrypted": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"private_key_fingerprint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceGoogleServiceAccountKeyCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	serviceAccount := d.Get("service_account_id").(string)
+
+	r := &iam.CreateServiceAccountKeyRequest{}
+
+	if v, ok := d.GetOk("key_algorithm"); ok {
+		r.KeyAlgorithm = v.(string)
+	}
+
+	if v, ok := d.GetOk("private_key_type"); ok {
+		r.PrivateKeyType = v.(string)
+	}
+
+	sak, err := config.clientIAM.Projects.ServiceAccounts.Keys.Create(serviceAccount, r).Do()
+	if err != nil {
+		return fmt.Errorf("Error creating service account key: %s", err)
+	}
+
+	if v, ok := d.GetOk("pgp_key"); ok {
```

Comment:
I'm a bit confused about this part, since it's just reading a value from the config, encrypting it, and then storing it back in state. I'm not totally against it, but it does make me cautious since it's not necessarily matching the API. What's the use case?

Reply:
Use case is to get GPG_key on output for using it with another tools. If it is possible to have them on outpup without storing it, i'm interesting.